### ISSUE TITLE
2:3 - "undefined method `each_index' for "":String"

### ISSUE THREAD

**Issue by marion-lv** [makersacademy/ruby-kickstart#228]
So, I was trying to take inspiration from 1:6 as it seemed to be asking for the same sort of actions, but I just keep getting the same error and I'm not sure how to fix it.

The error is:
1) String#every_other_char "" -> ""
     Failure/Error: expect("".every_other_char).to eq ""
     
     NoMethodError:
       undefined method `each_index' for "":String
     # ./session2/3-challenge/3_array.rb:10:in `every_other_char'
     # ./session2/4-spec/3.rb:3:in `block (2 levels) in <top (required)>'

And my code is:

class String
  def every_other_char
  self.split(",").values_at(* self.each_index.select {|i| i.even?}).join('')
  end
end

**Comment by tcpickard94** [comment]
Hi Marion. I believe it is failing because you are calling self.each_index. Since you are in the string class calling self on this is trying to invoke the method on the string. Trying splitting the string and saving that into a new variable, calling the variable instead of self (this way you are calling each_index on the array). Hope this helps

**Comment by marion-lv** [comment]
Hey Tom, that was really helpful, thanks!

Now my code looks like:

class String
  def every_other_char
  array = self.split(",")
  array.values_at(* array.each.select {|i| i.even?})
  end
end

and the error is:

1) String#every_other_char "" -> ""
     Failure/Error: expect("".every_other_char).to eq ""
     
       expected: ""
            got: []
     
       (compared using ==)
     # ./session2/4-spec/3.rb:3:in `block (2 levels) in <top (required)>'

Clearly it wants me to output "" when the array is empty rather than blank, any idea how I can get it to do that?

**Comment by marion-lv** [comment]
Thanks to Tom's help I've eventually managed to make my code work, so if anyone ever needs the fix:

```
class String
def every_other_char
array = self.split(",")
array.values_at(* array.each_index.select {|i| i.even?})
end
end
```
My array was returning `[]` instead of `""` when tested with a blank input, and Tom and I originally thought, it might be to do with my `.split(",")` method. We tried running `.split("")` instead, but no luck.

I decided to try something completely at random and added `.join("")` to make my code:

````
class String
def every_other_char
array = self.split(",")
array.values_at(* array.each_index.select {|i| i.even?}).join("")
end
end
```

I'm glad to report rake was happy with it!
